### PR TITLE
[REVIEW] Implement __cuda_array_interface__ for ingest

### PR DESCRIPTION
**Pull request by kkraus14**
Note: would need to install cupy, otherwise the test is skipped. Could switch this to PyTorch if wanted, but both are relatively large-ish projects so many not be desired outside of CI.

### CODE REVIEW

**Comment by kkraus14** [comment]
The tests could inadvertently pass by a cuda array-like object falling back to an iterator based approach instead of using the `__cuda_array_interface__` and would love ideas on how to better check for that.

**Comment by kkraus14** [comment]
Idea from @mrocklin: create a subclass of a cupy array or numba device array and delete the `__iter__` and any other attributes that would allow a fallback path to non `__cuda_array_interface__`

**Comment by kkraus14** [comment]
> Should cupy be added to the conda Dev environment config with this PR?

Yes it should, but I need to improve the testing to make sure it doesn't fall back to using an `__iter__` based method so I'll move this back to WIP.

**Comment by kkraus14** [comment]
In testing, CuPy arrays don't allow creation of numpy arrays or pyarrow arrays via standard creation methods so this is good to go when CI can pull in the changes to install CuPy.

**Comment by mike-wendt** [comment]
rerun tests

**Comment by kkraus14** [comment]
rerun tests

File: python/cudf/dataframe/columnops.py [hunk 1]
```diff
@@ -204,6 +205,18 @@ def as_column(arbitrary, nan_as_null=True):
                 mask = cudautils.mask_from_devary(arbitrary)
                 data = data.set_mask(mask)
 
+    elif cuda.is_cuda_array(arbitrary):
+        # Use cuda array interface to do create a numba device array by
+        # reference
+        new_dev_array = cuda.as_cuda_array(arbitrary)
```
Comment (thomcom) [inline python/cudf/dataframe/columnops.py]: `arbitrary` is already a cuda array - I don't like that you need `is_cuda_array` followed immediately by `as_cuda_array`. Are you sure that `as_cuda_array` is required here?

**Pull request merged**